app/main.py

```diff
@@ -1,3 +1,57 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def read_config_file(file_path: str) -> dict:
+    with open(file_path, "r") as file:
+        data = json.load(file)
+    return data
+
+
+def create_customers(customers_data: list) -> list:
+    customers = [
+        Customer(
+            customer_dict["name"],
+            customer_dict["product_cart"],
+            customer_dict["location"],
+            customer_dict["money"],
+            Car(
+                customer_dict["car"]["brand"],
+                customer_dict["car"]["fuel_consumption"]
+            )
+        ) for customer_dict in customers_data
+    ]
+    return customers
+
+
+def create_shops(shops_data: list) -> list:
+    shops = [
+        Shop(
+            shop_dict["name"],
+            shop_dict["location"],
+            shop_dict["products"]
+        ) for shop_dict in shops_data
+    ]
+    return shops
+
+
+def process_customers(customers: list, shops: list, fuel_cost: float) -> None:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]